src/context.js

```diff
@@ -0,0 +1,191 @@
+import { ControlFunction, HaltError } from './control';
+
+export class ExecutionContext {
```

Comment:
As mentioned in #61, I think `Task` might be a nice name for this?

Reply:
Let's hold off for now and see where it goes.